### PR TITLE
Optional arguments should advance the iterator in jerryx_arg_transform_optional

### LINKED ISSUE #2288: jerryx_arg_transform_optional does not advance iterator in case the value is undefined
I would expect in [this line] to pop the iterator if `js_arg` is `undefined`.
Otherwise there can only be a single optional argument. Another consequence is that  `jerryx_arg_transform_optional` does not work together with `jerryx_arg_transform_object_properties` because the iterator does not get advanced when an optional arg is `undefined`...

It looks like the unit test doesn't have either of these scenarios as a test case.

@jiangzidong do you agree this is a bug?

[this line]: https://github.com/jerryscript-project/jerryscript/blob/4e7a9d2d53804c1af7f7f7c57e2f6640898d0807/jerry-ext/arg/arg-transform-functions.c#L37

### PR DESCRIPTION
**Pull request by rerobika**
This patch fixes #2288

JerryScript-DCO-1.0-Signed-off-by: Robert Fancsik frobert@inf.u-szeged.hu

### CODE REVIEW

File: tests/unit-ext/test-ext-arg.c [hunk 1]
```diff
@@ -242,6 +245,52 @@ test_validator2_handler (const jerry_value_t func_obj_val __attribute__((unused)
...
+{
+
+  bool arg1 = false;
+
+  jerryx_arg_t mapping[] =
+  {
+    /* ignore this */
+    jerryx_arg_ignore (),
+    /* 1th argument should be boolean, and it is optional */
+    jerryx_arg_boolean (&arg1, JERRYX_ARG_COERCE, JERRYX_ARG_OPTIONAL),
```
Comment (galpeter) [inline tests/unit-ext/test-ext-arg.c]: AFAIK, there should be multiple optional args, to see if multiple optional arguments can be handled.

Comment (galpeter) [reply tests/unit-ext/test-ext-arg.c]: Oh and what if an "undefined" value is passed?

Comment (rerobika) [reply tests/unit-ext/test-ext-arg.c]: Good catch! I added tests for it.

**Comment by rerobika** [comment]
@galpeter Thanks for the review! I've updated the PR.

**Pull request merged**